xbmc/filesystem/NFSDirectory.cpp

```diff
@@ -266,25 +264,13 @@ bool CNFSDirectory::GetDirectory(const CURL& url, CFileItemList &items)
 
     iSize = tmpDirent.size;
     bIsDir = tmpDirent.type == NF3DIR;
-    lTimeDate = tmpDirent.mtime.tv_sec;
 
     if (!StringUtils::EqualsNoCase(strName,".") && !StringUtils::EqualsNoCase(strName,"..")
         && !StringUtils::EqualsNoCase(strName,"lost+found"))
     {
-      if(lTimeDate == 0) // if modification date is missing, use create date
-      {
-        lTimeDate = tmpDirent.ctime.tv_sec;
-      }
-
-      long long ll = lTimeDate & 0xffffffff;
-      ll *= 10000000ll;
-      ll += 116444736000000000ll;
-      fileTime.lowDateTime = (DWORD)(ll & 0xffffffff);
-      fileTime.highDateTime = (DWORD)(ll >> 32);
-      KODI::TIME::FileTimeToLocalFileTime(&fileTime, &localTime);
```

Comment:
Assuming that NFS file times are UTC then for Windows is still need

`FileTimeToLocalFileTime`  (now in CWIN32Util)

Reply:
One of the ideas of this whole PR is to change ibternal representation of `CDateTime` to UTC. So if the NFS time is UTC, then constructing CDateTime from it is proper. Did you check if current implementation produces wrong time?

Reply:
Yes, wrong file times because on Kodi File Manager is expected file times are same as OS file times. e.g. Synology NAS with NFS.

Is need apply timezone and DST offsets.

Reply:
With this diff is easy check if works:

```diff
diff --git a/xbmc/filesystem/NFSDirectory.cpp b/xbmc/filesystem/NFSDirectory.cpp
index 6d3588dccc..d51717879c 100644
--- a/xbmc/filesystem/NFSDirectory.cpp
+++ b/xbmc/filesystem/NFSDirectory.cpp
@@ -303,6 +303,11 @@ bool CNFSDirectory::GetDirectory(const CURL& url, CFileItemList &items)
       }
       pItem->SetPath(path);
       items.Add(pItem);
+
+      CLog::Log(LOGINFO, "'{}' {}-{}-{} - {}:{}:{}", pItem->GetPath(), pItem->m_dateTime.GetYear(),
+                pItem->m_dateTime.GetMonth(), pItem->m_dateTime.GetDay(),
+                pItem->m_dateTime.GetHour(), pItem->m_dateTime.GetMinute(),
+                pItem->m_dateTime.GetSecond());
     }
   }
```

```
2024-10-27 06:52:53.657 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/El Exorcista Del Papa (2023) - The Pope's Exorcist.mkv' 2023-6-23 - 18:16:8
2024-10-27 06:52:53.657 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Navidad en 8 Bits (2021) [4K].mkv' 2021-11-26 - 13:6:52
2024-10-27 06:52:53.657 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Reality (2023).mkv' 2023-9-27 - 16:48:38
2024-10-27 06:52:53.657 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/El proyecto Adam (2022) [4K, DV].mkv' 2022-5-19 - 13:29:12
2024-10-27 06:52:53.657 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/La niñas (2020).mkv' 2022-6-8 - 23:6:35
2024-10-27 06:52:53.657 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Englemageren (2023).mkv' 2024-2-25 - 13:31:59
2024-10-27 06:52:53.657 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Enola Holmes 2 (2022).mkv' 2022-11-5 - 10:40:10
2024-10-27 06:52:53.657 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Nuestro verano (2023).mkv' 2023-12-1 - 17:30:16
2024-10-27 06:52:53.657 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Jude (1996).mkv' 2023-9-7 - 19:10:58
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/La Ballena (2022) - The Whale.mkv' 2023-6-10 - 16:8:12
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Derribad al hombre (2019).mkv' 2022-7-9 - 12:28:38
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Bloodthirsty (2021).mkv' 2022-8-21 - 11:14:3
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/El arte de vivir bajo la lluvia (2019) [4K].mkv' 2021-10-11 - 21:34:17
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Flora y su hijo Max (2023).mkv' 2023-9-30 - 18:45:7
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Todos los nombres de Dios (2023).mkv' 2023-12-22 - 11:46:37
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/La Huérfana, Primer Asesinato (2022) - Orphan, First Kill.mkv' 2023-1-20 - 11:9:19
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Los buenos profesores (2023).mkv' 2024-9-13 - 17:33:24
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/1917 (2019) [4K, DV].mkv' 2023-9-11 - 14:14:21
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/El astronauta (2024) [4K, DV].mkv' 2024-5-9 - 19:44:58
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Godzilla Minus One (2023) [4K, DV].mkv' 2024-6-5 - 17:16:49
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/El Motin del Caine (2023).mkv' 2023-12-27 - 12:21:48
2024-10-27 06:52:53.658 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/First Cow (2019).mkv' 2024-7-31 - 13:53:53
2024-10-27 06:52:53.659 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Cinco lobitos (2022).mkv' 2022-11-15 - 11:8:19
2024-10-27 06:52:53.659 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Puñales por la espalda (2019) [4K].mkv' 2020-11-5 - 11:8:43
2024-10-27 06:52:53.659 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Desaparecidos en la noche (2024).mkv' 2024-7-14 - 9:25:7
2024-10-27 06:52:53.659 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/La Verdad Duele (2015) - Concussion.mkv' 2022-11-24 - 12:9:10
2024-10-27 06:52:53.659 T:7080     info <general>: 'nfs://192.168.50.13/volume1/NAS/MKV/2/Animales fantasticos y donde encontrarlos (2016) [4K].mkv' 2023-9-18 - 19:1:17
```

Reply:
Please come in slack #core if you have time! I dont see you among nicknames there…